source/AndroidResolver/src/PlayServicesResolver.cs

```diff
@@ -2191,15 +2192,45 @@ public static void SettingsDialog()
                 // https://docs.gradle.org/3.4/release-notes.html#the-java-library-plugin
                 // https://developer.android.com/studio/releases/gradle-plugin#3-0-0
                 var version = GradleVersion;
+                var versionComparer = new Dependency.VersionComparer();
                 var includeStatement =
                     !String.IsNullOrEmpty(version) &&
-                    (new Dependency.VersionComparer()).Compare("3.4", version) >= 0 ?
+                    versionComparer.Compare("3.4", version) >= 0 ?
                     "implementation" : "compile";
                 if (includeDependenciesBlock) lines.Add("dependencies {");
-                foreach (var packageSpecAndSources in GetPackageSpecs(dependencies: dependencies)) {
-                    lines.Add(String.Format(
-                        "    {0} '{1}' // {2}", includeStatement, packageSpecAndSources.Key,
-                        packageSpecAndSources.Value));
+                // Build a map of dependencies with the max version of that dependency.
+                // If different packages have different versions of the same dependency,
+                // we want to activate only the highest version but still include the
+                // other versions as commented out dependency lines.
+                var dependenciesMaxVersions = new Dictionary<string, string>();
+                foreach( var dependency in dependencies) {
+                    if(!dependenciesMaxVersions.ContainsKey(dependency.VersionlessKey))
```

Comment:
nit: Use `TryGetValue()` to prevent double look-up.
nit: always use bracket {} even there is one line of code unless it can be written in one line. Similarly in the other places.

Reply:
Done!